usdevs/uscwebsite-hackathon-backend
# Authentication via Telegram

This PR sets up the authentication aspect of the booking system and closes #14.

It accomplishes the following milestones:
- [x] Checks the authenticity of the telegram payload sent by the frontend
- [x] Sends the session token back to the client

It does not:
* check the database and populate the relevant tables

## How to test
1. Add a `BOT_TOKEN` field in your `.env` file.  Contact @misterpuffin for the token.
2. Go to the frontend website's login page (currently https://usdevs.github.io/uscwebsite-hackathon-backend/)
3. Login via telegram
4. The token should be displayed in the top box
![image](https://user-images.githubusercontent.com/53803800/206111586-1b27b273-8ba4-4c02-aa2a-680e94513fa8.png)
5. Access the main route `localhost:3000/` with an extra `Authentication` header in the http `GET` request with the token as the value.
How I did it is by sending the request via developer tools in Firefox
![image](https://user-images.githubusercontent.com/53803800/206112007-40ecdbee-2846-4455-96c2-ff0582f11c92.png)
6. The server should respond with response `200`.
